app/models/order_number/definition.rb

```diff
@@ -40,6 +40,14 @@ def id
       @id ||= quota_definition_sid
     end
 
+    def all_goods_nomenclatures
+      measures
+        .map(&:goods_nomenclature)
+        .compact
+        .uniq(&:goods_nomenclature_item_id)
+        .sort_by(&:goods_nomenclature_item_id)
```

Comment:
this piece of code is duplicated 4 times now,
it'd be great to dry this out, not sure which class should host that code for now,
maybe `MeasureCollection`? 🤔

Reply:
I'm not sure. There wasn't a very good place for it and the implementation is slightly different for footnotes so I ended up just leaving as is. I can move it to measure collection but it would break the common return signature of that class